exposure-notification/src/main/java/fi/thl/covid19/exposurenotification/FederationGatewayRestClientProperties.java

```diff
@@ -0,0 +1,46 @@
+package fi.thl.covid19.exposurenotification;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.boot.context.properties.ConstructorBinding;
+
+@ConfigurationProperties(prefix = "covid19.federation-gateway.rest-client")
+@ConstructorBinding
+public class FederationGatewayRestClientProperties {
+
+    public final TrustStore trustStore;
+    public final ClientKeyStore clientKeyStore;
+
+    public FederationGatewayRestClientProperties(TrustStore trustStore, ClientKeyStore clientKeyStore) {
+        this.trustStore = trustStore;
```

Comment:
Oh, one more thing: In dataobjects, use Objects.requireNonNull (I usually static import it) to make sure they can't be built wrong. Catches any mistakes as soon as possible.

Reply:
I'm not sure if it's necessary for objects managed by spring. It should take care of it and using it feels unnecessary. On every other object I share your opinion. Also, we could move to kotlin. :)

Reply:
Are you sure spring ensures it? I thought missing/null properties are still possible in configuration beans, but if they're not then sure, it's not necessary.

Reply:
I was, but I was wrong. :) In context of configuration properties missing properties will be initialized as null. Spring-way to achieve "not null" seems to be using Validator annotations for fields. This is quite odd that they have class level things like fail on unknown properties and things like that, but not null needs to be done field by field. So, I'll add some form. However, for normal beans it shouldn't be possible (if there is not a bug or something), or at least I can't easily imagine in what kind of situation this could be possible.

Reply:
Yeah, that is a bit odd. That's kind of why I mistrust all lib/framework code. For our own codebase, there are no nulls and that can be trusted. For everyone else's, it's easier to just verify everything "at the border" than dig under the hood every time to check. 

Also, I agree that we could move to kotlin - alas that was not my choice to make :)